neuror/zero_diameter_fixer.py

```diff
@@ -140,43 +138,35 @@ def fix_in_between(point, stack):
     if(len(stack) > 1 and
        stack[-2].section.diameters[stack[-2].point_id] <= SMALL and
        stack[-1].section.diameters[stack[-1].point_id] > SMALL):
-
+        count = 0
         for next_point in next_section_points_upstream(point):
+            count += 1
             if next_point.section.diameters[next_point.point_id] > SMALL:
-                connect_average(point, next_point)
+                connect_average(point, next_point, count)
                 break
 
     for next_point in next_section_points(point):
         fix_in_between(next_point, stack)
     stack.pop()
 
 
-def connect_average(point1, point2):
+def connect_average(start_point, stop_point, count):
     '''Apply a ramp diameter between the two points
 
     Re-implementation of https://bbpcode.epfl.ch/source/xref/sim/MUK/muk/Zero_Diameter_Fixer.cpp#232
     Contrary to the previous implementation the diameter the ramp is computed in term of
```

Comment:
is this docstring explaining the original differences between MUK and NeuroR?

Reply:
Seems so. thank you for pointing. I will delete it.

Reply:
But this means that it was not reproducing MUK on purpose, no? I don't quite follow the algorithm, but is it a big change? It does not look too harmful from the tests, only a slightly different interpolation, right?

Reply:
> it was not reproducing MUK on purpose, no?

Yes.

> I don't quite follow the algorithm, but is it a big change?

It's a major change. The test data didn't have cases that can exhibit such change.

Reply:
Ah ok, is there any way to know why this change was made for NeuroR? Maybe there was a good reason to depart from the original implementation?

Reply:
As the docstring says. This new implementation took into account path lengths. The original implementation bluntly distributed diameter among points. The new distributed them smarter. Segments with longer paths got bigger diameters. The only drawback is that we were not aware of this :D.

I propose to keep both implementations but use the original one by default.

Reply:
> I propose to keep both implementations but use the original one by default.

That's a good compromise, thanks.

Reply:
>> I propose to keep both implementations but use the original one by default.

> That's a good compromise, thanks.

@mgeplf Actually, I've made the opposite. By default the new one (smart with path lengths) is used. If you want to use the legacy (MUK) behavior then you need to pass the flag `legacy=True`. Please confirm/reject whether I should revert.

Reply:
sorry, should have been more clear; fine to have default be `legacy=False`